tensorflow_datasets/scripts/refactoring/refactoring_dataset.py

```diff
@@ -0,0 +1,148 @@
+  """Creates a new __init__.py. file"""
+  file_path = os.path.join(dest_path, '__init__.py')
+  dataset_name_cls = naming.snake_to_camelcase(dataset_info.dataset_name)
+  data = {'dataset_type': dataset_info.dataset_type,
+          'dataset_name': dataset_info.dataset_name,
+          'dataset_name_cls': dataset_name_cls}
+
+  with open(file_path, 'w') as f:
+    f.write(_INIT_FILE.format(**data))
+
+
+def copy_checksum_file(dataset_info, dest_path) -> None:
+  """Copy checksum.txt file"""
+  src_checksum_path = f'{URL_CHECKSUM_DIR}/{dataset_info.dataset_name}.txt'
+  if os.path.exists(src_checksum_path):
+    shutil.copy(src_checksum_path,
+                os.path.join(dest_path, f'{dataset_info.dataset_name}.txt'))
+
+
+def copy_make_data_file(dataset_info, dest_path) -> None:
+  """Copy fake data generation script file"""
+  src_fake_data_script_path = f'{FAKE_DATA_SCRIPT_DIR}/{dataset_info.dataset_name}.py'
+  if os.path.exists(src_fake_data_script_path):
+    shutil.copy(src_fake_data_script_path,
+                os.path.join(dest_path, 'make_fake_data.py'))
+
+
+def copy_fake_data_dir(dataset_info, dest_path) -> None:
+  """Copy fake data directory"""
+  src_fake_data_dir = os.path.join(FAKE_DATA_DIR, dataset_info.dataset_name)
+  if os.path.exists(src_fake_data_dir):
+    if os.path.exists(dest_path):
+      shutil.rmtree(dest_path)
+    shutil.copytree(src_fake_data_dir, os.path.join(dest_path, 'fake_data'))
+
+
+def copy_dataset_file(dataset_info, dest_path) -> None:
+  """Copy my_dataset.py file"""
+  shutil.copy(dataset_info.dataset_file,
+              os.path.join(dest_path, f'{dataset_info.dataset_name}.py'))
+
+
+def copy_dataset_test_file(dataset_info, dest_path) -> None:
+  """Copy my_dataset_test.py file"""
+  test_file = os.path.splitext(posixpath.basename(dataset_info.dataset_file))[0]
+  src_dataset_test_path = f'{dataset_info.dataset_path}/{test_file}_test.py'
+  if os.path.exists(src_dataset_test_path):
+    shutil.copy(src_dataset_test_path,
+                os.path.join(dest_path, f'{dataset_info.dataset_name}_test.py'))
+
+
+def extract_info(builder_cls: Type[tfds.core.DatasetBuilder]) -> BuilderInfo:
+  """Extract dataset name, filepath, type, path from the DatasetBuilder class."""
+  dataset_file = os.path.relpath(inspect.getfile(builder_cls))  # i.e tensorflow_datasets/image/mnist.py
+  dataset_name = builder_cls.name                 # name of the dataset i.e mnist
+  dataset_type = dataset_file.split('/')[1]       # type of the dataset i.e image
+  dataset_path = posixpath.split(dataset_file)[0] # i.e tensorflow_datasets/image
+  return BuilderInfo(dataset_file, dataset_name, dataset_type, dataset_path)
+
+
+def refactor_dataset(dataset_list):
+  """Refactoring all dataset into one folder"""
+  for dataset in dataset_list:
+    builder_cls = tfds.builder_cls(dataset)
+    dataset_info = extract_info(builder_cls)
+    # Refactored dataset path
+    refactor_dataset_path = os.path.join(NEW_TFDS_DIR,
```

Comment:
Datasets should stay in the same directory. Please remove the NEW_TFDS_DIR

Reply:
Done